File: apps/web/src/lib/format-localized-number.ts
/** Map Google Translate language codes to BCP 47 tags for Intl formatters. */
function resolveNumberLocale(lang: string | null | undefined): string {
  if (!lang || lang === 'en') return 'en-US';
  return lang;
}

export function formatLocalizedNumber(
  value: number,
  lang: string | null | undefined,
  options?: Intl.NumberFormatOptions,
): string {
  const locale = resolveNumberLocale(lang);
  try {
    return new Intl.NumberFormat(locale, options).format(value);
  } catch {
    return new Intl.NumberFormat('en-US', options).format(value);
  }
}

export function formatLocalizedInteger(
  value: number,
  lang: string | null | undefined,
): string {
  return formatLocalizedNumber(value, lang, {
    maximumFractionDigits: 0,
  });
}

File: apps/mobile/lib/format_localized_number.dart
import 'package:intl/intl.dart';

String resolveNumberLocale(String? lang) {
  if (lang == null || lang.isEmpty || lang == 'en') return 'en_US';
  return lang.replaceAll('-', '_');
}

String formatLocalizedNumber(
  num value,
  String? lang, {
  int? decimalDigits,
}) {
  final locale = resolveNumberLocale(lang);
  try {
    if (decimalDigits != null) {
      final pattern = decimalDigits == 0
          ? NumberFormat.decimalPattern(locale)
          : NumberFormat('#,##0.${'#' * decimalDigits}', locale);
      return pattern.format(value);
    }
    return NumberFormat.decimalPattern(locale).format(value);
  } catch (_) {
    return NumberFormat.decimalPattern('en_US').format(value);
  }
}

String formatLocalizedInteger(num value, String? lang) {
  return formatLocalizedNumber(value, lang, decimalDigits: 0);
}
